lib/tether/tether_linux.go

```diff
@@ -126,10 +137,16 @@ func (t *tether) childReaper() error {
 func (t *tether) stopReaper() {
 	defer trace.End(trace.Begin("Shutting down child reaping"))
 
-	// stop child reaping
-	log.Info("Shutting down reaper")
+	// Ordering is important otherwise we may one goroutine closing, and the other goroutine is trying to write afterwards
+	log.Debugf("Removing the signal notifier")
 	signal.Reset(syscall.SIGCHLD)
+
+	log.Debugf("Closing the channel done to reaper")
 	close(t.incoming)
+
+	log.Debugf("Sending true to the done channel")
+	// just closing the channel is not going to stop the iteration over the channel so signal it as well
```

Comment:
shouldn't `done` be closed first if incoming potentially has queued elements? It's the most immediate way of stopping reaping from continuing.
Of course calling code shouldn't be assuming reaping has stopped until this call returns, but...